### PR TITLE
Mijia refactoring

### PR DESCRIPTION
**Pull request by qwandor-google**
I've tried to separate the Mijia-specific code from the general BLE stuff. I've also added a few comments.

### CODE REVIEW

File: mijia/src/bluetooth.rs
Comment (alsuren) [inline mijia/src/bluetooth.rs:160]: I don't trust that this cast call is going to work when it encounters real data. I would be more comfortable if we had a test. Maybe deploy it to cottagepi and see if it works before merging?

I suppose I should have made a test for this as well.

File: mijia/src/bluetooth.rs
Comment (alsuren) [inline mijia/src/bluetooth.rs:19]: probably wants a doc comment with the word opaque in it somewhere.

(observation) I wonder if id is the right name. I'm not used to "id"s being opaque structs with private fields. You have more experience of languages with private fields than me, so I will defer to your judgement on this. I guess we can't use Handle because we're already using that for JoinHandles and it would become confusing.

File: mijia/src/lib.rs
Comment (alsuren) [inline mijia/src/lib.rs:31]:
```suggestion
```

File: mijia/src/lib.rs
Comment (alsuren) [inline mijia/src/lib.rs:46]: maybe make a `DeviceId::new(&str)` and do the clone in there?

File: mijia/src/lib.rs
Comment (alsuren) [inline mijia/src/lib.rs:99]:
```suggestion
    /// been connected, subscribe to notifications of temperature/humidity readings, and tweak the
    /// connection interval to save power.
```
I think we're increasing the interval, aren't we?

File: mijia/src/lib.rs
Comment (alsuren) [inline mijia/src/lib.rs:110]: your bluetooth.rs split is such a good idea that I'm suddenly trying to think of ways to move things like this into bluetooth.rs (definitely not in this PR).

File: publish-mqtt/src/main.rs
Comment (alsuren) [inline publish-mqtt/src/main.rs:303]:
```suggestion
    session.bt_session.start_discovery().await?;
```

File: publish-mqtt/src/main.rs
Comment (alsuren) [inline publish-mqtt/src/main.rs:337]: ❤️

File: mijia/src/bluetooth.rs
Comment (alsuren) [reply mijia/src/bluetooth.rs:160]: I put it on cottagepi with some debugging and debugs as `Variant(InternalArray { data: [48, 88, 91, 5, 1, 23, 33, 215, 56, 193, 164, 40, 1, 0], inner_sig: Signature("y") })` but then casts to None.

File: mijia/src/lib.rs
Comment (qwandor-google) [reply mijia/src/lib.rs:99]: You're quite right.

File: publish-mqtt/src/main.rs
Comment (qwandor-google) [reply publish-mqtt/src/main.rs:303]: Done.

File: mijia/src/lib.rs
Comment (qwandor-google) [reply mijia/src/lib.rs:46]: Done.

File: mijia/src/bluetooth.rs
Comment (qwandor-google) [reply mijia/src/bluetooth.rs:20]: Added a doc comment. Not sure what the best name is, I agree that it's a bit weird for something opaque to be called an ID.

Comment (alsuren) [reply mijia/src/bluetooth.rs:19]: It turns out that TypeId [from the standard library](https://doc.rust-lang.org/src/core/any.rs.html#417-419) is defined as:
```
#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
#[stable(feature = "rust1", since = "1.0.0")]
pub struct TypeId {
    t: u64,
}
```
so I think we're probably doing the rustic thing here.

File: mijia/src/bluetooth.rs
Comment (alsuren) [inline mijia/src/bluetooth.rs:169]: Pre-approved for when you push the change to use into_iter() etc. I really think that the best time to fix it properly is as part of #9 so I'm happy to leave it as-is for now.

If you want to write a test that's realistic, you will probably need to add an extra layer of nesting, so you can't Any your way out of trouble without boxed_clone, (we're currently a hashmap of {sv}, where the v is a Variant of &[u8] or something, so we're already borrowing from the outer hashmap)

**Review by alsuren — APPROVED** [review]

**Pull request merged**
